src/vs/workbench/api/browser/mainThreadCLICommands.ts

```diff
@@ -111,7 +111,9 @@ class RemoteExtensionCLIManagementService extends ExtensionManagementCLIService
 	}
 
 	protected override validateExtensionKind(manifest: IExtensionManifest, output: CLIOutput): boolean {
-		if (!this._extensionManifestPropertiesService.canExecuteOnWorkspace(manifest)) {
+		if (!this._extensionManifestPropertiesService.canExecuteOnWorkspace(manifest)
+			// Web extensions installed on remote can be run in web worker extension host
+			&& !this._extensionManifestPropertiesService.canExecuteOnWeb(manifest)) {
```

Comment:
Please add `isWeb` check because it is allowed only in web

Reply:
done :+1: